File: backend/app/services/evaluation.py
from agent import get_response
from netra import Netra
from uuid import uuid4

def run_evaluation(dataset_id: str) -> dict | None:
    dataset = Netra.evaluation.get_dataset(dataset_id) #type: ignore

    return Netra.evaluation.run_test_suite( #type: ignore
        name="Nova Single Turn",
        data=dataset,
        task=lambda message: get_response(message, thread_id=uuid4().hex)
    )

File: backend-ts/src/services/evaluation.ts
import { Netra } from "netra-sdk";
import { getResponse } from "../agent/index.js";
import { randomUUID } from "node:crypto";

export const runEvaluation = async (datasetId: string) => {
  const dataset = await Netra.evaluation.getDataset(datasetId);

  if (!dataset) {
    throw new Error("Dataset not found");
  }

  return await Netra.evaluation.runTestSuite(
    "Nova Single Turn",
    dataset as any, // Cast to any because the SDK's GetDatasetItemsResponse and Dataset interfaces might have slight mismatches in the d.ts
    async (message: string) =>
      await getResponse(message, randomUUID().replace(/-/g, "")),
  );
};
